src/pages/index.js

```diff
@@ -0,0 +1,291 @@
+      evt.preventDefault();
+    }
+    closeModal(popup);
+  });
+});
+
+profileEditButton.addEventListener("click", () => {
+  openModal(editModal);
+
+  nameInput.value = profileNameElement.textContent;
+  jobInput.value = profileJobElement.textContent;
+
+  profileSaveBtn.disabled = true;
+  profileSaveBtn.classList.add("modal__save-button_disabled");
+});
+
+// The form submission handler. Note that its name
+// starts with a verb and concisely describes what it does.
+function handleProfileFormSubmit(evt) {
+  // Prevent default browser behavior, see explanation below.
+  evt.preventDefault();
+
+  const newName = nameInput.value;
+  const newJob = jobInput.value;
+  profileSaveBtn.textContent = "Saving...";
+  api
+    .updateProfileInfo({ name: newName, about: newJob })
+    .then((data) => {
+      profileNameElement.textContent = data["name"];
+      profileJobElement.textContent = data["about"];
+    })
+    .catch((err) => {
+      console.error(err);
+    })
+    .finally(() => {
+      closeModal(editModal);
+      profileSaveBtn.textContent = "Save";
+    });
+}
+
+// Connect the handler to the form, so it will watch for the submit event.
+profileFormElement.addEventListener("submit", handleProfileFormSubmit);
+
+const cardsList = document.querySelector(".cards__list");
+const cardTemplate = document.querySelector("#card");
+
+let selectedCard;
+let selectedCardId;
+
+function getCardElements(data) {
+  const cardElement = cardTemplate.content.cloneNode(true);
+  const likeButton = cardElement.querySelector(".card__button");
+  const deleteButton = cardElement.querySelector(".card__delete-button");
+
+  const cardImage = data["link"];
+  const cardName = data["name"];
+  const cardImageElement = cardElement.querySelector(".card__image");
+
+  cardImageElement.src = cardImage;
+  cardImageElement.alt = cardName;
+  cardElement.querySelector(".card__name").textContent = cardName;
+
+  deleteButton.addEventListener("click", (evt) => {
+    handleDeleteCard(evt, data["_id"]);
+  });
+
+  if (data.isLiked) {
+    likeButton.classList.add("card__button_liked");
+  }
+
+  likeButton.addEventListener("click", (evt) => {
+    handleLike(evt, data["_id"]);
+  });
+
+  cardImageElement.addEventListener("click", () => {
+    photoModalImage.src = cardImage;
+    photoModalImage.alt = cardName;
+    photoModalCaption.textContent = cardName;
+    openModal(photoModal);
+  });
+  return cardElement;
+}
+
+const postAddButton = document.querySelector(".profile__new-btn");
+const addModal = document.querySelector("#add-modal");
+
+const postFormElement = document.forms["add-post"];
+const postSaveBtn = postFormElement.querySelector(".modal__save-button");
+
+const linkInput = postFormElement.querySelector("#link");
+const captionInput = postFormElement.querySelector("#caption");
+
+postAddButton.addEventListener("click", () => {
+  openModal(addModal);
+  postSaveBtn.disabled = true;
+  postSaveBtn.classList.add("modal__save-button_disabled");
```

Comment:
You should create function `disableButton` in `validation.js` for disabling not to duplicate the code. Then you need to import it into `index.js`